pkg/controller/migplan/validation.go

```diff
@@ -206,6 +223,68 @@ func (r ReconcileMigPlan) validateNamespaces(plan *migapi.MigPlan) error {
 		})
 		return nil
 	}
+	limit := Settings.Plan.NsLimit
+	if count > limit {
+		message := fmt.Sprintf(NsLimitExceededMessage, limit)
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     NsLimitExceeded,
+			Status:   True,
+			Reason:   LimitExceeded,
+			Category: Critical,
+			Message:  message,
+		})
+		return nil
+	}
+
+	return nil
+}
+
+// Validate the number of running pods (limit) within each namespace.
+func (r ReconcileMigPlan) validatePodLimit(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended, NsLimitExceeded) {
+		plan.Status.StageCondition(PodLimitExceeded)
+		return nil
+	}
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	limit := Settings.Plan.PodLimit
+	for _, name := range plan.GetSourceNamespaces() {
+		list := kapi.PodList{}
+		options := k8sclient.InNamespace(name)
+		err := client.List(context.TODO(), options, &list)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+		count := 0
```

Comment:
Does this count get set back to 0 for each namespace? My impression was that we wanted to limit the total number of pods being migrated rather than # pods per namespace.

Reply:
You're right.  I did not remember that correctly.  Fixing.

Reply:
Fixed.